packages/@glimmer/interfaces/lib/runtime/debug-render-tree.d.ts

```diff
@@ -3,7 +3,12 @@ import type { SimpleElement, SimpleNode } from '@simple-dom/interface';
 import type { Bounds } from '../dom/bounds.js';
 import type { Arguments, CapturedArguments } from './arguments.js';
 
-export type RenderNodeType = 'outlet' | 'engine' | 'route-template' | 'component';
+export type RenderNodeType =
+  | 'outlet'
+  | 'engine'
+  | 'route-template'
+  | 'component'
+  | 'remote-element';
```

Comment:
I think it would make more sense to add a more generic type here, either `keyword` or `block` (as in `{{#this-is-a-block}}`). This would, in theory, allow it to be used in the future for other things like `{{#if}}`, `{{#each}}`, etc. 

Not saying we should add those now – with the current architecture of the debug render tree it's not obvious that we can afford to do it, both in terms of the added (unstrippable) code size and runtime cost. I am ok with added these things that are high value (improving the UX in the inspector that is otherwise very annoying to infer from within the inspector code) and happens infrequently, like `{{#in-element}}`, but, I would otherwise be quite uneasy about doing it for other keywords (and probably helpers, which presumably happens a lot more than modifiers).

We inherited the current UX and architecture of that tab in the inspector from wayyyyy back. DebugRenderTree was a massive improvement compared to the code that existed before in terms of rationalizing the data structure and processing, but IMO the architecture itself (where the code lives, how to make it more opt-in, the amount of work it takes after each re-render, and the serializing/sending/deserializing of the tree) is itself overdue for a ground-up rethink, especially in light of Embroider/code-stripping. So we are a bit bottlenecked in terms of what kinds of improvements we can be making that are still viable within the limits of the current architecture.

That being said, I think there is no reason to limit ourselves here and lock ourselves in here. Hence the suggestion. I am imagining that when the inspector code sees the `keyword`/`block` type node, it can just always render something like `{{#${node.name} ${node.args.positional} ${node.args.named}}}`. If we pull on this thread more we'll eventually have to figure out what to do with block params (if we can even do anything about it), but we can worry about that another time.

Reply:
used keyword